hepstats/hypotests/calculators/frequentist_calculator.py

```diff
@@ -0,0 +1,425 @@
+                    floating parameters in the sample Default is `hepstats.fitutils.sampling.base_sampler`.
+                sample : function used to get samples from the sampler.
+                    Default is `hepstats.fitutils.sampling.base_sample`.
+
+            Example with `zfit`:
+                >>> import zfit
+                >>> from zfit.core.loss import UnbinnedNLL
+                >>> from zfit.minimize import MinuitMinimizer
+
+                >>> obs = zfit.Space('x', limits=(0.1, 2.0))
+                >>> data = zfit.data.Data.from_numpy(obs=obs, array=np.random.normal(1.2, 0.1, 10000))
+                >>> mean = zfit.Parameter("mu", 1.2)
+                >>> sigma = zfit.Parameter("sigma", 0.1)
+                >>> model = zfit.pdf.Gauss(obs=obs, mu=mean, sigma=sigma)
+                >>> loss = UnbinnedNLL(model=[model], data=[data], fit_range=[obs])
+
+                >>> calc = FrequentistCalculator(input=loss, minimizer=MinuitMinimizer(), ntoysnull=1000, ntoysalt=1000)
+        """
+
+        super(FrequentistCalculator, self).__init__(input, minimizer)
+
+        self._toysnull = {}
+        self._toysalt = {}
+        self._ntoysnull = ntoysnull
+        self._ntoysalt = ntoysalt
+        self._sampler = sampler
+        self._sample = sample
+        self._toys_loss = {}
+
+    @property
+    def ntoysnull(self):
+        """
+        Returns the number of toys loss for the null hypothesis.
+        """
+        return self._ntoysnull
+
+    @property
+    def ntoysalt(self):
+        """
+        Returns the number of toys loss for the alternative hypothesis.
+        """
+        return self._ntoysalt
+
+    def sampler(self, floating_params=None):
+        """
+        Create sampler with models.
+
+        Args:
+            floating_params (list): floating parameters in the sampler
+
+        Example with `zfit`:
+            >>> sampler = calc.sampler(floating_params=[zfit.Parameter("mean")])
+        """
+        self.set_dependents_to_bestfit()
+        nevents = []
+        # self.model and self.data defined in BaseCalculator
+        for m, d in zip(self.model, self.data):
+            if m.is_extended:
+                nevents.append("extended")
+            else:
+                nevents.append(get_nevents(d))
+
+        return self._sampler(self.model,  nevents, floating_params)
+
+    def sample(self, sampler, ntoys, parameter=None, value=None):
+        """
+        Returns the samples generated from the sampler for a given value of a parameter
+
+        Args:
+            sampler (list): generator of samples
+            ntoys (int): number of samples to generate
+            parameter (optional): floating parameter in the sampler
+            value (optional): value of the parameter
+
+        Example with `zfit`:
+            >>> mean = zfit.Parameter("mean")
+            >>> sampler = calc.sampler(floating_params=[mean])
+            >>> sample = calc.sample(sampler, 1000, mean, 1.2)
```

Comment:
Looking at the signature of the method, and the docstring, I did not immediately realise why I could not simply have `sample(self, sampler, ntoys, **parameter)`. After all, if all you want is a parameter name and value, then `sample(sampler, 1000, mean=1.2)` would do. But in fact you need a specific object type, as seen from the example. Then this needs to be explained in the docstring under Args I reckon. (Apologies if I'm being thick and this is all obvious for a user of the package, which I'm not yet.)

Reply:
In fact I could put a `POI` object here which contains the `parameter` and a `value`. Would that make more sense to you?

Reply:
Yep, seems good.